quesma/plugins/ingest_validator/ingest_validator.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+package ingest_validator
+
+import (
+	"quesma/clickhouse"
+	"quesma/logger"
+	"quesma/plugins"
+	"quesma/quesma/config"
+	"quesma/quesma/types"
+	"quesma/schema"
+	"quesma/util"
+	"reflect"
+)
+
+type IngestValidator struct {
+	cfg            config.QuesmaConfiguration
+	schemaRegistry schema.Registry
+	table          string
+	tableMap       clickhouse.TableMap
+}
+
+func (iv *IngestValidator) Transform(document types.JSON) (types.JSON, error) {
+	clickhouseTable, ok := iv.tableMap.Load(iv.table)
+	if !ok {
+		logger.Error().Msgf("Table %s not found", iv.table)
+		return document, nil
+	}
+
+	for k, v := range document {
+		if v != nil {
+			column := clickhouseTable.Cols[k]
+			if column != nil {
+				kind, _ := util.KindFromString(column.Type.String())
+				if kind != reflect.TypeOf(v).Kind() {
+					//logger.Error().Msgf("Field %s has wrong type %s, expected %s", k, getTypeName(v), clickhouseTable.Cols[k].Type.String())
+					return document, nil
+				}
+			}
+		}
+	}
+	return document, nil
+}
+
+func (iv *IngestValidator) ApplyIngestTransformers(table string, cfg config.QuesmaConfiguration, schema schema.Registry, tableMap clickhouse.TableMap, transformers []plugins.IngestTransformer) []plugins.IngestTransformer {
+	transformers = append(transformers, &IngestValidator{cfg: cfg, schemaRegistry: schema, table: table, tableMap: tableMap})
+	return transformers
+}
```

Comment:
It's a plugin now. I'm not sure about it.

Reply:
So, I made some changes to avoid cyclic dependencies like moving some interfaces around, we should rethink the whole plugin architecture and dependencies

Reply:
Especially dependencies.